src/nvim/fileio.c

```diff
@@ -4514,9 +4515,14 @@ int vim_rename(char_u *from, char_u *to)
     if (STRLEN(from) >= MAXPATHL - 5)
       return -1;
     STRCPY(tempname, from);
-    for (n = 123; n < 99999; ++n) {
-      sprintf((char *)path_tail(tempname), "%d", n);
-      if (!os_file_exists(tempname)) {
+    for (n = 123; n < 99999; n++) {
+      // the 'n' in snprintf is calculated by subtracting the difference
+      // between the tail pointer and the start of the array from the
+      // total buffer length
+      char * tail = (char *)path_tail(tempname);
+      snprintf(tail, (MAXPATHL + 1) - (tail - (char *)tempname - 1), "%d", n);
```

Comment:
the comment is too long, and the mention of "n" is confusing given the presence of the `n` parameter which is unrelated to the 'n' mentioned in the comment. Better to leave the comment out, it mostly restates what the code is doing.